lib/plugins/aws/package/compile/events/apiGateway/index.js

```diff
@@ -264,6 +264,35 @@ class AwsCompileApigEvents {
           );
         }
 
+        if (
+          this.serverless.service.provider.name === 'aws' &&
+          Object.values(this.serverless.service.functions).some(({ events }) =>
+            events.some(({ http }) => {
+              if (http && http.authorizer) {
+                if (_.isObject(http.authorizer)) {
+                  // Deprecation should not apply to already existing, external authorizers
+                  if (http.authorizer.authorizerId) return false;
+                  // Deprecation should not apply to `aws_iam` authorization type
+                  if (http.authorizer.type && http.authorizer.type.toUpperCase() === 'AWS_IAM') {
+                    return false;
+                  }
+                  if (http.authorizer.identitySource) return false;
+
+                  return true;
+                }
+                // Authorizer is defined as a string and we want to check if its `aws_iam` - in such case the deprecation will not apply
+                return !(http.authorizer.toUpperCase() === 'AWS_IAM');
+              }
+              return false;
+            })
+          )
+        ) {
+          this.serverless._logDeprecation(
+            'AWS_API_GATEWAY_DEFAULT_IDENTITY_SOURCE',
+            'Starting with `v3.0.0`, `identitySource` property for `http.authorizer` will no longer be set to "method.request.header.Authorization" by default. If you want to keep this setting, please set it explicitly in your configuration.'
```

Comment:
I think there are two issues we deal with:

1. We attempt to set default value to `identitySource` (design bug)
2. We treat `undefined` and `null` as set for `identitySource` differently (logic bug)

Probably best way to fix it, is to leave how it works now, and fix both of the issues with next major.

Still the way the deprecation afaik should be communicated is:

```
Starting with v3 not setting "functions[].http.authorizer.identitySource" will no longer result with "method.request.header.Authorization" being as as default.
If you want to keep this setting, please set it explicitly in your configuration. If you don't want this to be set, set it to "null"
```